File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char, int> character_map;

        int left = 0;
        int maxLength = 0;

        for (int right = 0; right < s.length(); right++) {

            character_map[s[right]]++;

            while (character_map[s[right]] > 1) {
                character_map[s[left]]--;
                left++;
            }

            maxLength = max(maxLength, right - left + 1);
        }

        return maxLength;
    }
};

File: Data Structures & Algorithms/valid-palindrome-ii/submission-1.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        reversed_String = s[::-1]
        count = 0

        if (reversed_String == s):
            return True
        
        left = 0
        right = len(s)-1

        while (left <= right):
            if (s[left] == s[right]):
                bFlag = True
            else:
                count += 1
        
            left+=1
            right-=1

        if (count == 1):
            return True
        else:
            return False


        